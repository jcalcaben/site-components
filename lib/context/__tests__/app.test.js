import React, { useEffect } from 'react';
import AppContextProvider, { useAppContext } from '../app'

import TestRenderer, { act } from 'react-test-renderer'

const log = jest.fn();

const Consumer = jest.fn(() => {
    const contextValues = useAppContext();

    useEffect(() => {
        log(contextValues)
    }, [contextValues])

    return <span />
})

test('render children', () => {
    const Component = AppContextProvider;

    const testRenderer = TestRenderer.create(<Component><span>Test</span></Component>)

    expect(testRenderer.toJSON()).toMatchSnapshot();
})

test('provide values to consumer', () => {
    const Component = AppContextProvider;

    let testRenderer;

    act(() => {
        testRenderer = TestRenderer.create(
            <Component><Consumer /></Component>
        );
    });

    expect(log).toHaveBeenCalledTimes(1);

    expect(log).toHaveBeenNthCalledWith(1, [expect.objectContaining({
        "activePanel": undefined
    }), {}]);

})