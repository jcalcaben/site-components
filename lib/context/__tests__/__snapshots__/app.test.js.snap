// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render children 1`] = `
<span>
  Test
</span>
`;
